### PR TITLE
Finally ports TG drinks

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Ports TG Drinks:
## Why It's Good For The Game
TG drinks are simply better.
This also means we can just use their wiki as reference
## Changelog
:cl:
add: Ports TG Drinks
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm [hunk 1]
```diff
@@ -0,0 +1,912 @@
...
+	name = "Ice"
+	description = "Frozen water, your dentist wouldn't like you chewing this."
+	reagent_state = SOLID
+	color = "#619494" // rgb: 97, 148, 148
+	taste_description = "ice"
+	default_container = /obj/item/reagent_containers/cup/glass/ice
+	adj_temp = - 7
+	targ_temp = BODYTEMP_NORMAL - 15
+
+/datum/reagent/consumable/soy_latte
```
Comment (Looking4bros) [inline code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:408]: Our versions of Coffee Latte and Soy Latte are duplicated on line 818 and 830, theirs doesn't have the Healing effects that ours does, as well as ours are childed/branced under Coffee. (muh power game)

File: code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm [hunk 2]
```diff
@@ -0,0 +1,912 @@
...
+/// Simple healing proc to avoid copy paste
+/datum/reagent/consumable/proc/milk_heal(mob/living/L, metabolism)
+	L.heal_limb_damage(0.2)
+	if(holder.has_reagent(/datum/reagent/consumable/capsaicin))
+		holder.remove_reagent(/datum/reagent/consumable/capsaicin, 2)
+
+/datum/reagent/consumable/milk/on_mob_life(mob/living/L, metabolism)
+	milk_heal(L, metabolism)
+	return ..()
+
```
Comment (TiviPlus) [inline code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:157]: erm?

File: code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:408]: respond to above comment pls

File: code/modules/reagents/chemistry/recipes/food_drink.dm [hunk 3]
```diff
@@ -76,442 +67,322 @@
...
 	results = list(/datum/reagent/consumable/ethanol/whiskey_cola = 3)
-	required_reagents = list(/datum/reagent/consumable/ethanol/whiskey = 2, /datum/reagent/consumable/drink/cold/space_cola = 1)
+	required_reagents = list(/datum/reagent/consumable/ethanol/whiskey = 2, /datum/reagent/consumable/space_cola = 1)
 
-/datum/chemical_reaction/screwdriver
-	name = "Screwdriver"
+/datum/chemical_reaction/screwdriverrecipe
 	results = list(/datum/reagent/consumable/ethanol/screwdrivercocktail = 3)
-	required_reagents = list(/datum/reagent/consumable/ethanol/vodka = 2, /datum/reagent/consumable/drink/orangejuice = 1)
+	required_reagents = list(/datum/reagent/consumable/ethanol/vodka = 2, /datum/reagent/consumable/orangejuice = 1)
```
Comment (TiviPlus) [inline code/modules/reagents/chemistry/recipes/food_drink.dm:146]: fo these need to be migrated?

File: code/modules/reagents/reagent_containers/cups/_cup.dm [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+	)
+
+	///Like Edible's food type, what kind of drink is this?
+	var/drink_type = NONE
+	///The last time we have checked for taste.
+	var/last_check_time
+	///How much we drink at once, shot glasses drink more.
+	var/gulp_size = 5
+	///Whether the 'bottle' is made of glass or not so that milk cartons dont shatter when someone gets hit by it.
+	var/isGlass = FALSE
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/_cup.dm:21]: trr camelcase

File: code/modules/reagents/reagent_containers/cups/_cup.dm [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+	drink_type = style.drink_type
+
+/// Callback for [datum/component/takes_reagent_appearance] to reset to no foodtypes
+/obj/item/reagent_containers/cup/proc/on_cup_reset()
+	drink_type = NONE
+
+/*
+/obj/item/reagent_containers/cup/beaker
+	name = "beaker"
+	desc = "A beaker. It can hold up to 50 units."
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/_cup.dm:137]: Whats going on here?

File: code/modules/reagents/reagent_containers/cups/bottle.dm [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+	possible_transfer_amounts = list(5, 10)
+	amount_per_transfer_from_this = 5
+	///variable to tell if the bottle can be refilled
+	var/cap_on = TRUE
+
+/obj/item/reagent_containers/cup/bottle/syrup_bottle/examine(mob/user)
+	. = ..()
+	. += span_notice("Alt-click to toggle the pump cap.")
+	. += span_notice("Use a pen on it to rename it.")
+	return
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/bottle.dm:183]:
```suggestion
```

File: code/modules/reagents/reagent_containers/cups/bottle.dm [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+	. = ..()
+	cap_on = !cap_on
+	if(!cap_on)
+		icon_state = "syrup_open"
+		balloon_alert(user, "removed pump cap")
+	else
+		icon_state = "syrup"
+		balloon_alert(user, "put pump cap on")
+	update_icon_state()
+	return
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/bottle.dm:222]:
```suggestion
```

File: code/modules/reagents/reagent_containers/cups/drinks.dm [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+/obj/item/reagent_containers/cup/glass/coffee
+	name = "robust coffee"
+	desc = "Careful, the beverage you're about to enjoy is extremely hot."
+	icon = 'icons/obj/drinks/coffee.dmi'
+	icon_state = "coffee"
+	base_icon_state = "coffee"
+	list_reagents = list(/datum/reagent/consumable/coffee = 30)
+	isGlass = FALSE
+	drink_type = BREAKFAST
+	var/lid_open = 0
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/drinks.dm:41]: this is a bool

File: code/modules/reagents/reagent_containers/cups/drinks.dm [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+
+/obj/item/reagent_containers/cup/glass/coffee/no_lid
+	icon_state = "coffee_empty"
+	list_reagents = null
+
+/obj/item/reagent_containers/cup/glass/coffee/examine(mob/user)
+	. = ..()
+	. += span_notice("Alt-click to toggle cup lid.")
+	return
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/drinks.dm:50]:
```suggestion
```

File: code/modules/reagents/reagent_containers/cups/drinks.dm [hunk 10]
```diff
@@ -0,0 +1,375 @@
...
+
+/obj/item/reagent_containers/cup/glass/coffee/AltClick(mob/user)
+	. = ..()
+	lid_open = !lid_open
+	update_icon_state()
+	return
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/drinks.dm:56]:
```suggestion
	update_icon_state()
```

File: code/modules/reagents/reagent_containers/cups/drinks.dm [hunk 11]
```diff
@@ -0,0 +1,375 @@
...
+	var/cap_icon = 'icons/obj/drinks/drink_effects.dmi'
+	var/cap_icon_state = "bottle_cap_small"
+	var/cap_on = TRUE
+	var/cap_lost = FALSE
+	var/mutable_appearance/cap_overlay
+	var/flip_chance = 10
+
+/obj/item/reagent_containers/cup/glass/waterbottle/Initialize(mapload)
+	cap_overlay = mutable_appearance(cap_icon, cap_icon_state)
+	. = ..()
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/drinks.dm:144]:
```suggestion
	return ..()
```

File: code/modules/reagents/reagent_containers/cups/drinks.dm [hunk 12]
```diff
@@ -0,0 +1,375 @@
...
+	if(cap_on)
+		cap_on = FALSE
+		animate(src, transform = null, time = 2, loop = 0)
+		to_chat(user, span_notice("You remove the cap from [src]."))
+		playsound(loc, 'sound/effects/can_open1.ogg', 50, TRUE)
+	else
+		cap_on = TRUE
+		to_chat(user, span_notice("You put the cap on [src]."))
+	update_appearance()
+	return
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/drinks.dm:175]:
```suggestion
```

File: code/modules/reagents/reagent_containers/cups/glassbottle.dm [hunk 13]
```diff
@@ -0,0 +1,688 @@
...
+	force = 15 //Smashing bottles over someone's head hurts.
+	throwforce = 15
+	var/broken_worn_icon_state = "broken_beer"
+	worn_icon_list = list(
+		slot_l_hand_str = 'icons/mob/inhands/items/drinks_lefthand.dmi',
+		slot_r_hand_str = 'icons/mob/inhands/items/drinks_righthand.dmi'
+		)
+	drink_type = ALCOHOL
+	///Directly relates to the 'knockdown' duration. Lowered by armor (i.e. helmets)
+	var/bottle_knockdown_duration = BOTTLE_KNOCKDOWN_DEFAULT_DURATION
```
Comment (TiviPlus) [inline code/modules/reagents/reagent_containers/cups/glassbottle.dm:37]: new vars after old ones

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm (hunk 2 above)
Comment (Looking4bros) [reply code/modules/reagents/chemistry/reagents/drinks/drink_reagents.dm:157]: This is just consolidation of all the 0.2 brute heal that all milk and milk childed reagents have. Seems TG did that as they separated the childed reagents into their own, all this is. Dunno if it'd be better some other way.

File: code/modules/reagents/chemistry/recipes/food_drink.dm (hunk 3 above)
Comment (Xander3359) [reply code/modules/reagents/chemistry/recipes/food_drink.dm:146]: Yes

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Can shatter is not a bool properly still but lets see if anything breaks meanwhile

**Comment by Looking4bros** [comment]
Hearty Punch doesn't have any effects, it's so joever...

**Pull request merged**